.github/workflows/publish.yml

```diff
@@ -11,34 +11,73 @@ on:
     paths:
       - "src/**"
       - ".github/workflows/publish.yml"
+  repository_dispatch:
+    types: [create-release]
+    client_payload:
+      release_tag:
+        required: true
 
 jobs:
+  create_release:
+    if: ${{ github.event_name == 'repository_dispatch' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: Brightspace/third-party-actions@actions/checkout
+      - name: Create release
+        env:
+          RELEASE_TAG: ${{ github.event.client_payload.release_tag }}
+        run: |
+          gh release create "$RELEASE_TAG" --draft `
+            --title "Release $RELEASE_TAG" `
+            --notes "Release notes for version $RELEASE_TAG"
+
   build:
     strategy:
       matrix:
         include:
           - machine: windows-latest
             platform: win
+            file_name: D2L.Bmx.exe
           - machine: macos-latest
             platform: osx
+            file_name: D2L.Bmx
     runs-on: ${{ matrix.machine }}
     timeout-minutes: 10
+    env:
+      PLATFORM: ${{ matrix.platform }}
     steps:
       - name: checkout
         uses: Brightspace/third-party-actions@actions/checkout
       - name: publish
         shell: pwsh
+        working-directory: src/D2L.Bmx
+        run: |
+          dotnet publish -r $env:PLATFORM-x64 -c Release -o build
+      - name: upload build
+        if: ${{ github.event_name == 'repository_dispatch' }}
+        shell: bash
         env:
+          RELEASE_TAG: ${{ github.event.client_payload.release_tag }}
           PLATFORM: ${{ matrix.platform }}
+          FILE_NAME: ${{ matrix.file_name }}
+          GH_TOKEN: ${{ github.token }}
         working-directory: src/D2L.Bmx
-        run: dotnet publish -r $env:PLATFORM-x64 -c Release -o out
+        run: |
+          cd build
+          mv "$FILE_NAME" "$PLATFORM-$FILE_NAME"
+          gh release upload "$RELEASE_TAG" "$PLATFORM-$FILE_NAME"
 
   build_docker:
+    if: ${{ always() }}
+    needs: create_release
```

Comment:
```suggestion
```

This is meant to go but related to [this ](https://github.com/Brightspace/bmx/pull/260/files#r1176898984) comment and what I was doing before